core/iwasm/compilation/aot_emit_numberic.c

```diff
@@ -816,7 +816,15 @@ is_targeting_soft_float(AOTCompContext *comp_ctx, bool is_f32)
          * doesn't have or enable Floating-Point Coprocessor Option on xtensa. */
         ret = (!is_f32 || strstr(feature_string, "-fp")) ? true : false;
     else if (is_target_riscv(comp_ctx))
-        ret = !strstr(feature_string, "+d") ? true : false;
+        /*
+         * Note: Use builtin intrinsics since hardware float operation
+         * will cause rodata relocation, this will try to use hardware
+         * float unit (by return false) but handled by software finally
+         */
+        if (comp_ctx->disable_llvm_intrinsics)
+            ret = false;
+        else
+            ret = !strstr(feature_string, "+d") ? true : false;
```

Comment:
Put this code block in to  { } to avoid compiler warning

Reply:
Done